components/brave_wallet/browser/swap_service.cc

```diff
@@ -293,11 +302,18 @@ void SwapService::GetPriceQuote(mojom::SwapParamsPtr swap_params,
   auto internal_callback =
       base::BindOnce(&SwapService::OnGetPriceQuote,
                      weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+
+  base::flat_map<std::string, std::string> request_headers;
+  const std::string zero_ex_api_key = GetZeroExAPIKey();
+  if (!zero_ex_api_key.empty()) {
+    request_headers = {{"0x-api-key", zero_ex_api_key}};
+  }
+
   api_request_helper_.Request(
       "GET",
       GetPriceQuoteURL(std::move(swap_params),
                        json_rpc_service_->GetChainId(mojom::CoinType::ETH)),
-      "", "", true, std::move(internal_callback));
+      "", "", true, std::move(internal_callback), request_headers);
```

Comment:
Could you please make that an anonymous function like CommonSwapHeaders, so no copy-paste below

Reply:
Fixed in https://github.com/brave/brave-core/pull/18033.